www/js/lib/zimfile.js

```diff
@@ -110,6 +110,42 @@ define(['xzdec_wrapper', 'util', 'utf8', 'q', 'zimDirEntry'], function(xz, util,
         }
     };
 
+    /**
+     * Reads the whole MIME type list and returns it as a populated Map
+     * The mimeTypeMap is extracted once after the user has picked the ZIM file
+     * and is stored as ZIMFile.mimetypes.
+     * 
+     * @returns {Promise} A promise for the MIME Type list as a Map
+     */
+    ZIMFile.prototype._mimeTypeMap = function() {
+        var typeMap = new Map;
+        return this._readSlice(this.mimeListPos, 256).then(function(data) {
+            // DEV: We have read 256 bytes: increase this if you encounter longer MIME type lists
+            // also change "while (pos < 255)" below
+            if (data.subarray) {
+                var i = 1;
+                var pos = -1;
+                var mimeString;
+                while (pos < 255) {
+                    pos++; 
+                    mimeString = utf8.parse(data.subarray(pos), true);
+                    // If the parsed data is an empty string, we have reached the end of the MIME type list, so break 
+                    if (!mimeString) break;
+                    // Store the parsed string in the Map
+                    typeMap.set(i, mimeString);
+                    i++;
+                    while (data[pos]) {
```

Comment:
This is safer than `while (data[pos] !== 0)` because if `pos` increments up to 256 (beyond the end of our data slice), then `data[pos]` will return `undefined`. If we just test `!==0` then we could end up with an infinite loop.

Reply:
That's right.
Don't hesitate to copy the algorithm used in the libzim as it is the reference implementation

Reply:
OK. It's worth noting that there are several other places where `while (data[pos] !== 0)` is used in our code, which seems to me less safe (and more verbose) than `whlle (data[pos])`